src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -153,16 +153,13 @@ private static Vector256<float> MultiplyAdd(Vector256<float> src1, Vector256<flo
         // Multiply matrix times vector into vector.
         public static unsafe void MatMul(AlignedArray mat, AlignedArray src, AlignedArray dst, int crow, int ccol)
         {
-            Contracts.Assert(src.Size == dst.Size);
```

Comment:
Aren't there still asserts we can do here (and in all the functions you are removing the asserts for)?  like `dst.Size >= crow` and `src.Size < ccol`?

Reply:
I guess those kinds of asserts should go into `src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs`